src/openzeppelin/mocks/reentrancy_mock.cairo

```diff
@@ -0,0 +1,85 @@
+#[abi]
+trait IReentrancyGuardAttacker {
+  fn call_sender();
+}
+
+#[abi]
+trait IReentrancyGuard {
+    fn count_this_recursive(n: felt252);
+}
+
+#[contract]
+mod ReentrancyMock {
+    // OZ modules
+    use openzeppelin::security::reentrancyguard::ReentrancyGuard;
+    use openzeppelin::utils::check_gas;
+
+    // Dispatchers
+    use super::IReentrancyGuardAttackerDispatcher;
+    use super::IReentrancyGuardAttackerDispatcherTrait;
+    use super::IReentrancyGuardDispatcher;
+    use super::IReentrancyGuardDispatcherTrait;
+
+    // Other
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+    use starknet::get_contract_address;
+
+    struct Storage {
+        _counter: felt252
+    }
+
+    #[constructor]
+    fn constructor(initial_number: felt252) {
+        _counter::write(initial_number);
+    }
+
+    #[view]
+    fn current_count() -> felt252 {
+        _counter::read()
+    }
+
+    #[external]
+    fn callback() {
+        ReentrancyGuard::start();
+        count();
+        ReentrancyGuard::end();
+    }
+
+    #[external]
+    fn count_local_recursive(n: felt252) {
+        ReentrancyGuard::start();
+        check_gas();
+        if n != 0 {
+            count();
+            count_local_recursive(n - 1);
+        }
+        ReentrancyGuard::end();
+    }
+
+    #[external]
+    fn count_this_recursive(n: felt252) {
```

Comment:
```suggestion
    fn count_this_recursive(n: felt252) {
```
should we rename this to something a bit more telling? like `count_external_recursive` or something

Reply:
Yeah, that's a much better name